packages/lit-dev-content/site/docs/frameworks/react.md

```diff
@@ -0,0 +1,176 @@
+  }}
+/>
+```
+
+### How it works
+
+During a render, the wrapper receives props from React and based on the options and the custom element class, changes the behavior of some of the props:
+
+* If a prop name is a property on the custom element, as determined with an `in` check, the wrapper sets that property on the element to the prop value
+* If a prop name is an event name passed to the `events` option, the prop value is passed to `addEventListener()` with the name of the event.
+* Otherwise the prop is passed to React's `createElement()` to be rendered as an attribute.
+
+Both properties and events are added in `componentDidMount()` and `componentDidUpdate()` callbacks, because the element must have already been instantiated by React in order to access it.
+
+For events, `createComponent()` accepts a mapping of React event prop names to events fired by the custom element. For example passing `{onFoo: 'foo'}` means a function passed via a prop named `onFoo` will be called when the custom element fires the `foo` event with the event as an argument.
+
+## useController
+
+Reactive controllers allow developers to hook in to a component's lifecycle to bundle
+together state and behavior related to a feature. They are similar to React
+hooks in the user cases and capabilities, but are plain JavaScript objects
+instead of functions with hidden state.
+
+`useController()` lets you make React hooks out of reactive controllers allowing for the sharing of state and behaviors across web components and React.
+
+### Usage
+
+```jsx
+import React from 'react';
+import {useController} from '@lit-labs/react/use-controller.js';
+import {MouseController} from '@example/mouse-controller';
+
+// Write a custom React hook function:
+const useMouse = () => {
+  // Use useController to create and store a controller instance:
+  const controller = useController(React, (host) => new MouseController(host));
+  // Return relevant data for consumption by the component:
+  return controller.pos;
+};
+
+// Now use the new hook in a React component:
+const Component = (props) => {
+  const mousePosition = useMouse();
+  return (
+    <pre>
+      x: {mousePosition.x}
+      y: {mousePosition.y}
+    </pre>
+  );
+};
+```
+
+See the [mouse controller example](../../composition/controllers/#example:-mousemovecontroller) in the reactive controller docs for its implementation.
+
+### How it works
+
+`useController()` uses `useState` to create and store an instance of a controller and a `ReactControllerHost`. It then calls the controller's lifecycle from the hook body and `useLayoutEffect` callbacks, emulating the `ReactiveElement` lifecycle as closely as possible. `ReactControllerHost` implements `addController` so that controller composition works and nested controller lifecycles are called correctly. `ReactControllerHost` also implements `requestUpdate` by calling a `useState` setter, so that a controller with new renderable state can cause its host component to re-render.
```

Comment:
This section needs some help here. It's a little too blunt.

```suggestion
`useController()` creates a custom host object for the controller passed to it and drives the controller's lifecycle by using React hooks.

- `useState()` is used to store an instance of a controller and a `ReactControllerHost`
- The hook body and `useLayoutEffect()` callbacks emulate the `ReactiveElement` lifecycle as closely as possible.
- `ReactControllerHost` implements `addController()` so that controller composition works and nested controller lifecycles are called correctly.
- `ReactControllerHost` also implements `requestUpdate()` by calling a `useState()` setter, so that a controller can cause its host component to re-render.
```